xpra/codecs/nvenc/encoder.pyx

```diff
@@ -2321,15 +2568,34 @@ cdef class Encoder:
             r = self.functionList.nvEncEncodePicture(self.context, &picParams)
         raiseNVENC(r, "flushing encoder buffer")
 
-    def compress_image(self, device_context, image, int quality=-1, int speed=-1, options=None, int retry=0):
+    def compress_image(self, device_context, image, int quality=-1, int speed=-1, options=None, int retry=0, unsigned long unique_id=0):
         assert device_context, "no cuda device context"
         #cuda_device_context.__enter__ does self.context.push()
+        log("called compress_image with common encoder: %s, context: %#x, frames: %i", self, <uintptr_t> self.context, self.frames)
+        log("compress_image%s", (quality, speed, options, retry))
+
+        #FIXME: do not lock encode_lock on USE_SINGLETON_ENCODER disabled
```

Comment:
Actually, I think we do want a lock in all cases.
With multiple concurrent users each having their own encode thread, this could easily break nvenc.
(but the lock would need to protect more than just `compress_image` in that case)

Reply:
I was assuming that multiple threads with their own cuda context and encoder would not need a global lock holding them (this is a global/module level var created) and could operate in parallel without stepping over eachother - the only reason why this is here is to ensure that parallel threads don't step on eachother as they're writing/reading from the same buffers/memory locations (in the singleton case).
Do you suspect that we'd need a global lock in the case where a singleton isn't used?